src/data/entity_registry.ts

```diff
@@ -0,0 +1,50 @@
+import { HomeAssistant } from "../types";
+import computeStateName from "../common/entity/compute_state_name";
+
+export interface EntityRegistryEntry {
+  entity_id: string;
+  name: string;
+  platform: string;
+  config_entry_id?: string;
+  device_id?: string;
+  disabled_by?: string;
+}
+
+export interface EntityRegistryEntryUpdateParams {
+  name: string;
+  new_entity_id: string;
+}
+
+export const computeEntityRegistryName = (
+  hass: HomeAssistant,
+  entry: EntityRegistryEntry
+) => {
+  if (entry.name) {
+    return entry.name;
+  }
+  const state = hass.states[entry.entity_id];
+  return state ? computeStateName(state) : null;
+};
+
+export const fetchEntityRegistry = (hass: HomeAssistant) =>
```

Comment:
```suggestion
export const fetchEntityRegistry = (hass: HomeAssistant): Promise<EntityRegistryEntry[]> =>
```

Reply:
TypeScript is in this case actually smart enough to take the return type form `hass.callWS`